### PR TITLE
Add debug-only diagnostics to TagSource

### PR DESCRIPTION
**Pull request by Pilchie**

### CODE REVIEW

**Comment by Pilchie** [comment]
@jasonmalinowski @CyrusNajmabadi Thoughts?

**Comment by CyrusNajmabadi** [comment]
:+1:

**Comment by balajikris** [comment]
:+1:

File: src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_ReferenceCounting.cs [hunk 1]
```diff
@@ -17,7 +17,14 @@ private partial class TagSource
             {
                 if (!Environment.HasShutdownStarted)
                 {
-                    Contract.Fail(@"Should have been disposed!");
+#if DEBUG
+                    Contract.Fail($@"Should have been disposed!
+
+{_stackTrace}
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_ReferenceCounting.cs]: Might want to say "creating stack trace" just so it's obvious what that is.

**Comment by CyrusNajmabadi** [comment]
BTW, something else useful to add would be if you captured the stacktrace when the owning XXXTaggerProvider was created.  Then have the tagsource print out both its stack trace and it's owner's stack trace. That way we could tell:
1. Is this a classification tagger or a squiggle tagger?
2. What test triggered this problem, and what was going on that caused it.

With all the asynchrony in the system, the stack trace for when the tagger got created is currently not that helpful.  

If you're finished with this, i can add that functionality.

**Comment by Pilchie** [comment]
Good idea. You go ahead, I already merged this PR.

**Pull request merged**